hazelcast/src/test/java/com/hazelcast/query/impl/OrResultSetTest.java

```diff
@@ -0,0 +1,109 @@
+package com.hazelcast.query.impl;
+
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.query.QueryException;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class OrResultSetTest {
+
+    @Test
+    public void size() {
+        int size = 100000;
+        Set<QueryableEntry> entries1 = generateEntries(size);
+        Set<QueryableEntry> entries2 = generateEntries(size);
+        List<Set<QueryableEntry>> indexedResults = new ArrayList<Set<QueryableEntry>>();
+        indexedResults.add(entries1);
+        indexedResults.add(entries2);
+
+        OrResultSet resultSet = new OrResultSet(indexedResults);
+
+        int sizeMethod = resultSet.size();
+        int sizeIterator = 0;
+        for (QueryableEntry queryableEntry : resultSet) {
+            sizeIterator++;
+        }
+
+        assertEquals(sizeIterator, 2 * size);
```

Comment:
I think this should be `assertEquals(2 * size, sizeIterator);`, since the order is `expected` then `actual` value. Same for the next assertion.